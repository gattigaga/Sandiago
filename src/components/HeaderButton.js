import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import { TouchableNativeFeedback } from "react-native";
import Icon from "react-native-vector-icons/Ionicons";

const Container = styled.View`
  width: 48px;
  height: 72px;
  justify-content: center;
  align-items: center;
`;

const StyledIcon = styled(Icon)`
  font-size: 32px;
  color: white;
`;

const HeaderButton = ({ icon, onPress }) => (
  <TouchableNativeFeedback onPress={onPress}>
    <Container>
      <StyledIcon name={icon} />
    </Container>
  </TouchableNativeFeedback>
);

HeaderButton.propTypes = {
  icon: PropTypes.string,
  onPress: PropTypes.func
};

export default HeaderButton;
