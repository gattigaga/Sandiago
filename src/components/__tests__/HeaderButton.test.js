import React from "react";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";

import HeaderButton from "../HeaderButton";

describe("HeaderButton", () => {
  const setup = propOverrides => {
    const props = {
      icon: "ios-settings",
      onPress: jest.fn(),
      ...propOverrides
    };

    const wrapper = shallow(<HeaderButton {...props} />);

    return {
      wrapper,
      props
    };
  };

  it("should renders default", () => {
    const { wrapper } = setup();
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it("should calls 'onPress' while pressed", () => {
    const { wrapper, props } = setup();

    wrapper.simulate("press");

    expect(props.onPress).toBeCalled();
  });
});
