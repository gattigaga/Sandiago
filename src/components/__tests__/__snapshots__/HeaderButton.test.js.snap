// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderButton should renders default 1`] = `
<DummyTouchableNativeFeedback
  onPress={[MockFunction]}
>
  <Styled(Component)>
    <Styled(Icon)
      name="ios-settings"
    />
  </Styled(Component)>
</DummyTouchableNativeFeedback>
`;
